### PR TITLE
Use watir and remove support for watir-webdriver

### PR DESCRIPTION
**Pull request by tnypxl**
PR Notes:

#### Watir Gem
Watir 6.x.x deprecates `watir-webdriver` and has improved APIs. Taza will now use the new `watir` gem by default. This change also removes the need for Selenium and Firewatir dependencies.

#### A note to monkey patchers
`Taza::Browser` has been simplified to defer to watir and selenium-webdriver exclusively for browser support. That means existing monkey patches of `Taza::Browser.create_watir_webdriver` will break.

### CODE REVIEW

File: lib/taza/browser.rb [hunk 1]
```diff
@@ -6,64 +6,29 @@ class Browser
     # Example:
     #     browser = Taza::Browser.create(Taza::Settings.config)
     #
-    def self.create(params={})
-      self.send("create_#{params[:driver]}".to_sym,params)
-    end
-
-    def self.browser_class(params)
-      self.send("#{params[:driver]}_#{params[:browser]}".to_sym)
+    def self.create(params={}, *args)
```
Comment (rrosenblum) [inline lib/taza/browser.rb]: Typically, options hashes are the last parameter to a method.

File: lib/taza/browser.rb [hunk 2]
```diff
@@ -6,64 +6,29 @@ class Browser
...
-      Watir::Safari.new
-    end
-
-    def self.watir_ie(params)
-      require 'watir'
-      if params[:attach]
-        browser = Watir::Browser.find(:title, //)
-      end
-      browser || Watir::Browser.new
+      Selenium::WebDriver.for(params[:browser].to_sym, *args)
```
Comment (rrosenblum) [inline lib/taza/browser.rb]: If browser always needs to be a symbol, should it be converted before the method calls that use it?

File: lib/taza/browser.rb (hunk 1 above)
Comment (tnypxl) [inline lib/taza/browser.rb]: I'm going to remove `*args`. In order for *args to even get used, you'd have to pass it in on Taza::Site initialization. I realized that might be especially confusing given your point.

File: lib/taza/browser.rb (hunk 2 above)
Comment (tnypxl) [inline lib/taza/browser.rb]: Thats a good idea. Thanks!

**Comment by tnypxl** [comment]
@rrosenblum re: Ruby 2.4.1 compatibility: Thats probably out of scope for this PR. A PR that resolves issues with 2.4.1 would probably be preferred. /cc @orieken @uchagani

**Comment by rrosenblum** [comment]
> Ruby 2.4.1 compatibility: Thats probably out of scope for this PR.

Yes, official 2.4 support should be done in a different PR. I should have been more clear with my initial comment. I tried running the tests using Ruby 2.4, and they blew up because of `watir-webdriver`. There is a good chance that Taza will work with Ruby 2.4 after this is merged in.

**Comment by tnypxl** [comment]
@rrosenblum Those 2.4.1 issues should go away then with switching to watir as a dependency.

**Comment by tnypxl** [comment]
Some more changes to the browser class: https://github.com/tnypxl/taza/blob/browser_defs/lib/taza/browser.rb

Suggestions welcomed.

**Pull request closed without merging**